docs/ldml/tr35-numbers.md

```diff
@@ -1315,6 +1317,54 @@ Here is a set of heuristic rules that may be helpful:
 * A currency symbol in the input should be interpreted as the longest match found in the set of possible currency symbols.
 * Especially in cases of ambiguity, the user's input should be echoed back, properly formatted according to the locale, before it is actually used for anything.
 
+## 8 <a name="Number_Range_Formatting" href="#Number_Range_Formatting">Number Range Formatting</a>
+
+Often ranges of numbers are presented to users, such as in “Length: 3.2–4.5 centimeters”. This means any length from 3.2 cm to 4.5 cm, inclusive.
+
+To format a number range, the following steps are taken:
+
+1. Format the lower bound and the upper bound independently following the steps in <a name="Number_Format_Patterns" href="#Number_Format_Patterns">Number Format Patterns</a>.
+1. If the resulting values are identical, stop evaluating these steps and, instead, perform the steps in <a name="Approximate Number Formatting" href="#Approximate Number Formatting">Approximate Number Formatting</a>.
+    1. Note: This behavior may be customized in order to, for example, print the range despite the endpoints being identical. However, a spec-compliant implementation must support approximate number formatting.
+1. Perform the steps in <a name="Collapsing_Number_Ranges" href="#Collapsing_Number_Ranges">Collapsing Number Ranges</a>, obtaining modified *lower* and *upper* values.
+1. Obtain a number range pattern by following the steps in <a name="Range_Pattern_Processing" href="#Range_Pattern_Processing">Range Pattern Processing</a>.
+1. Substitute *lower* as `{0}` and *upper* as `{1}` into the range pattern from the previous step.
+
+For plural rule selection of number ranges, see <a name="Plural_Ranges" href="#Plural_Ranges">Plural Ranges</a>.
+
+### 8.1 <a name="Approximate_Number_Formatting" href="#Approximate_Number_Formatting">Approximate Number Formatting</a>
+
+TODO
+
+### 8.2 <a name="Collapsing_Number_Ranges" href="#Collapsing_Number_Ranges">Collapsing Number Ranges</a>
+
+*Collapsing* a number range refers to the process of removing duplicated information in the *lower* and *upper* values. For example, if the lower string is "3.2 centimeters" and the upper string is "4.5 centimeters", it is desirable to remove the extra "centimeters" token.
```

Comment:
This is not going to be easily implementable, since programs would have to check for the ill-defined concept of semantic equivalents. Hard to do across languages.

To make this work in practice, I think it needs to go a bit lower in the process. One approach (you can probably think of a better): when the number is being formatted, we need to get the "measure" in the broad sense, including currencies, and any compact decimal prefix/suffix, and their locations in the resulting string (relative to themselves and the number. EG, 

<3.5, number> <M, compact-designation> <centimeters, id:centimeter>
vs
<M, compact-designation> <3.5, number> <centimeters, id:centimeter>
vs
...
Then we can describe what to do.

Reply:
Fixed.